src/runtime/containerd-shim-v2/service.go

```diff
@@ -301,7 +301,7 @@ func trace(ctx context.Context, name string) (otelTrace.Span, context.Context) {
 }
 
 func (s *service) Cleanup(ctx context.Context) (_ *taskAPI.DeleteResponse, err error) {
-	span, _ := trace(s.rootCtx, "Cleanup")
+	span, spanCtx := trace(s.rootCtx, "Cleanup")
```

Comment:
What is the rule of spanning trace context here? I see that some places are using previously saved context, others are using the passed down `ctx`. We need to unify them.

Reply:
@bergwolf - when we receive requests from containerd (or crio if they pass contexts, I'm not sure), they are tied to the life of the particular task service function or to the shim itself.

Based on that, we have a context that has notion of being tied to lifecycle of the sandbox, as well as contexts which are tied to a particular request.

I think it makes sense to use the context from the specific request whenever possible, so we can have spans that run from containerd down to the agent in the guest. However, we'll need to make sure to use the long running ctx for functions which are dependent on it (IIRC -we pass a context to ttRPC?)

Reply:
@bergwolf in containerd-shim-v2 I added rootCtx so that all of the trace spans are organized under a single root span, which is associated with a root context. This was done based on feedback in previous issues and PRs that people wanted spans organized this way. (We don't actually need to do this; we  could use the contexts passed into calls in service.go. It will just change the organization of them so that they are not grouped under a single root span.) 

Aside from create.go and service.go, where tracing is started and higher level spans are created, for tracing we typically want to use a context passed to the function to get properly ordered spans. 

I agree with @egernst here. I avoided changing the use of a service or shim struct if it was used for anything other than tracing, so if there are instances of this it is a bug.